developer-docs-site/docs/guides/move-guides/move-scripts.md

```diff
@@ -0,0 +1,145 @@
+---
+title: "Move Scripts"
+slug: "move-scripts"
+---
+
+# Move Scripts
+
+You can use Move scripts to conduct arbitrary tasks on chain. 
+
+## Example use case
+
+The following example calls functions on the [aptos_coin.move](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/aptos_coin.move) module to confirm the balance of the destination account is less than `desired_balance`, and if so, tops it up to `desired_balance`.
+
+```move
+script {
+    use std::signer;
+    use aptos_framework::aptos_account;
+    use aptos_framework::aptos_coin;
+    use aptos_framework::coin;
+
+    fun main(src: &signer, dest: address, desired_balance: u64) {
+        let src_addr = signer::address_of(src);
+
+        let balance = coin::balance<aptos_coin::AptosCoin>(src_addr);
+        if (balance < desired_balance) {
+            aptos_account::transfer(src, dest, desired_balance - balance);
+        };
+
+        addr::my_module::do_nothing();
+    }
+}
+```
+
+This will end up in a distinct Move script file. Notice also how it calls a function in a Move module defined as:
+```
+module addr::my_module {
+    public entry fun do_nothing() { }
+}
+```
+
+That entry should go in a Move module file.
+
+Now you need to determine:
+
+- Where do I put these files?
+- What do I name them?
+- Do I need a `Move.toml`?
+- How do I run my script with the CLI?
+
+## Execution
+
+Let us run through how to execute a Move script with a step-by-step example using the [Aptos CLI](../../cli-tools/aptos-cli-tool/use-aptos-cli.md).
+
+1. Make a new directory for your work:
+```sh
+mkdir testing
+cd testing
+```
+
+2. Set up the Aptos CLI:
+```sh
+aptos init
+```
+The CLI will ask you which network you want to work with (e.g. `devnet`, `testnet`, `mainnet`). It will also ask you for your private key (which looks like this: `0xf1adc8d01c1a890f17efc6b08f92179e6008d43026dd56b71e7b0d9b453536be`), or it can generate a new one for you, as part of setting up your account.
+
+3. From this same directory, initialize a new Move project:
+```sh
+aptos move init --name my_script
+```
+
+4. Create a `top_up.move` file containing the example script above in a `sources/` subdirectory of your `testing/` directory.
+
+5. Create a `my_module.move` Move module file containing the `do_nothing()` function above in the same  `sources/` subdirectory.
+
+6. Create a `Move.toml` file in the root of your `testing/` directory containing:
+
+```
+[package]
+name = 'my_script'
+version = '1.0.0'
+
+[dependencies.AptosFramework]
+git = 'https://github.com/aptos-labs/aptos-core.git'
+rev = 'main'
```

Comment:
This should use `devnet` or another branch.  We probably need to cleanup all uses of `main`

Reply:
I am replacing `main` with `devnet` here.